private/buf/cmd/buf/command/alpha/plugin/pluginpush/pluginpush.go

```diff
@@ -376,7 +384,19 @@ func findExistingDigestForImageID(ctx context.Context, plugin *bufpluginconfig.C
 		return "", err
 	}
 	auth := &authn.Basic{Username: authConfig.Username, Password: authConfig.Password}
-	tags, err := remote.List(repo, remote.WithContext(ctx), remote.WithAuth(auth))
+	remoteOpts := []remote.Option{remote.WithContext(ctx), remote.WithAuth(auth)}
+	// First attempt to see if the current image digest matches the image ID
```

Comment:
This is a lot faster in the case of the image already existing in the BSR with the same digest as before. It only has to check one digest (the current one) and not potentially search through dozens of tags.